src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -126,15 +138,36 @@ public static IResourceBuilder<AzureStorageResource> RunAsEmulator(this IResourc
         builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(builder.Resource, async (@event, ct) =>
         {
             var connectionString = await builder.Resource.GetBlobConnectionString().GetValueAsync(ct).ConfigureAwait(false);
-
-            if (connectionString == null)
+            if (connectionString is null)
             {
-                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
+                throw new DistributedApplicationException($"BeforeResourceStartedEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
             }
 
             blobServiceClient = CreateBlobServiceClient(connectionString);
         });
 
+        builder.ApplicationBuilder.Eventing.Subscribe<ResourceReadyEvent>(builder.Resource, async (@event, ct) =>
+        {
+            var connectionString = await builder.Resource.GetBlobConnectionString().GetValueAsync(ct).ConfigureAwait(false);
+            if (connectionString is null)
+            {
+                throw new DistributedApplicationException($"ResourceReadyEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
+            }
+
+            if (blobServiceClient is null)
+            {
+                throw new DistributedApplicationException($"BlobServiceClient was not created for the '{builder.Resource.Name}' resource.");
+            }
+
+            foreach (var blobStorageResources in builder.Resource.Blobs)
+            {
+                foreach (var blobContainer in blobStorageResources.BlobContainers)
+                {
+                    await blobServiceClient.GetBlobContainerClient(blobContainer.Name).CreateIfNotExistsAsync(cancellationToken: ct).ConfigureAwait(false);
```

Comment:
I notice the other resource event hooks don't do any kind of exception handling in these creation codepaths but should we be? What do the failure modes for these look like?

Reply:
According to the [docs](https://learn.microsoft.com/dotnet/api/azure.storage.blobs.blobcontainerclient.createifnotexistsasync?view=azure-dotnet#remarks) `CreateIfNotExistsAsync` can throw [RequestFailedException](https://learn.microsoft.com/dotnet/api/azure.requestfailedexception?view=azure-dotnet). But it's likely it'll fail before getting here anyway (e.g., while accessing the storage or the blobs resources).

Reply:
Yes, letting the exception bubble out of this call is good. It'll correctly log and fail dependent resources.